### PR TITLE
Lower requested resorces for sidecar containers

### LINKED ISSUE #508: Consider lowering resource requests for sidecars or make them configurable

### PR DESCRIPTION
**Pull request by AMecea**
closes/fixes #508 

---
 - [x] I've made sure the [Changelog.md](https://github.com/presslabs/mysql-operator/blob/master/Changelog.md) will remain up-to-date after this PR is merged.

### CODE REVIEW

**Comment by biwwy0** [comment]
I have a PR ready to define this values in podSpec, which we are running in production now. As to our use case particularly metrics-exporter is susceptible to huge memory usage (thousands of small databases with userstat enabled). Why is this PR here?

**Comment by AMecea** [comment]
Hi @biwwy0, we are using (in production) one database per cluster so we determined that the limits that we just set are fine. We are not aware of what is the usage of the exporter container with lots of databases. I will update this PR to let metrics exporter resource be configurable (as you mentioned) and don't set by default any limit for metrics container. Is this ok for you?

Thank you for noticing it!

**Comment by biwwy0** [comment]
Hi @AMecea, I can prepare a PR for you for that. If you guide me! Thanks

**Comment by AMecea** [comment]
I appreciate that, but I'm in a hurry I want to finish release 0.4 today. I will appreciate your code review on this! Also, if you have something prepared please open a PR!

**Comment by stankevich** [comment]
I also have reservations about having non-configurable resource requests/limits as a long-term solution. We don't currently set CPU limits on any of our workloads (due to https://github.com/kubernetes/kubernetes/issues/51135 and https://github.com/kubernetes/kubernetes/issues/67577), so we resort to removing the CPU limits from the operator. It would be ideal if requests/limits were configurable for all containers from the cluster spec. Thanks!

File: pkg/apis/mysql/v1alpha1/mysqlcluster_defaults.go [hunk 1]
```diff
@@ -68,6 +68,20 @@ func (c *MysqlCluster) setPodSpecDefaults(spec *PodSpec) {
...
+			corev1.ResourceMemory: resource.MustParse("32Mi"),
+		}
+	}
+
+	if len(spec.MySQLOperatorSidecarResources.Requests) == 0 {
+		spec.MySQLOperatorSidecarResources.Requests = corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("10m"),
+			corev1.ResourceMemory: resource.MustParse("64Mi"),
+		}
+	}
```
Comment (AMecea) [inline pkg/apis/mysql/v1alpha1/mysqlcluster_defaults.go]: @calind what do you think about setting only requested? What about chosesn default values?

Limit is not set, the user can set those values by themselves.

Comment (calind) [reply pkg/apis/mysql/v1alpha1/mysqlcluster_defaults.go]: I would be hesitant to let the sidecars consume _unlimited_ amount of resources. As long as users have a way to configure the limit I would enforce a sensible default. This goes into `0.4` release so it's ok to modify the defaults.

File: Changelog.md [hunk 2]
```diff
@@ -53,6 +55,9 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
  * Update cron documentation
  * Set InnoDB buffer parameter: `innodb_buffer_pool_instances` to `min(resources.limit.cpu,
    floor(innodb_buffer_pool_size/1G))` (see #502)
+ * Change requested resources for sidecar containers: `sidecar`: (requested=10m/32Mi,
+   limit=`.spec.podSpec.rsources.limit`); `exporter`: (requested=10m/32Mi. limit=100m/128Mi);
+   `heartbeat`: (requested=10m/32Mi, limit=100m/64Mi)
```
Comment (calind) [inline Changelog.md]:
```suggestion
 * Change default resource requests/limits for `sidecar` container: requested=10m/32Mi, limit= the same as `.spec.podSpec.resources.limit`
 * Change default resource requests/limits for `exporter` container: requested=10m/32Mi, limit=100m/128Mi
 * Change default resource requests/limits for `heartbeat` container: requested=10m/32Mi, limit=100m/64Mi
```

**Pull request merged**